src/Microsoft.AspNetCore.Razor.Language/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -574,20 +575,16 @@ public void WriteTagHelperRuntime(CodeRenderingContext context, DefaultTagHelper
             }
             else if (node is CSharpCodeIntermediateNode)
             {
-                var error = new RazorError(
-                    LegacyResources.TagHelpers_CodeBlocks_NotSupported_InAttributes,
-                    SourceLocation.FromSpan(span),
-                    span == null ? -1 : span.Value.Length);
-                context.Diagnostics.Add(RazorDiagnostic.Create(error));
+                var location = span.HasValue ? span.Value : SourceSpan.Undefined;
```

Comment:
This could be simplified to `span ?? SourceSpan.Undefined` and then just inlined.